src/utils/marks/marks.js

```diff
@@ -95,16 +95,22 @@ export class Mark {
 
         const rects = Array.from(this.range.getClientRects());
         const stringRects = rects.map((r) => JSON.stringify(r));
-        const stringRectsSet = new Set(stringRects);
-        const rectsSet = Array.from(stringRectsSet).map((sr) => JSON.parse(sr));
-
-        return rectsSet.reduce((filteredRects, curRect) => {
-            const shouldNotPush = rectsSet.some((item) => curRect !== item && contains(curRect, item));
+        const stringRectsSet = new Set(stringRects); // de-duplicate the boxes
+        const uniqueRects = Array.from(stringRectsSet)
+            .map((sr) => JSON.parse(sr))
+            .filter(r => r.width !== 0); // filter out any boxes with a width of 0
```

Comment:
Filtering out rectangles with a width of 0 here prevents the rect with a width of zero from being chosen over the rect with an actual width in the `reduce()` below due to the `contains()` check.

<img width="533" alt="Screen Shot 2022-11-22 at 12 30 37 PM" src="https://user-images.githubusercontent.com/6599979/203381974-b00b1405-56f2-4f17-9231-fa32918b5773.png">